Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+        let userId: any = request.body.user.account_id
+        if (isNaN(userId)) {
+            response.status(500).json("Invalid search params entered");
+        }
+        else {
+            try {
+                let arrayOfData = await this.dataSetService.getUserSavedDatasets(userId)
+                return response.status(200).send(arrayOfData);
+            } catch (err) {
+                response.status(500).send(err);
+            }
+        }
+    }
+
+    //TODO: Readd later
+    /**
+     * This controller will take a request, grab the user email and data set ID, verify said ID is a number,
+     * and then send the service a request to add the user's saved data set preference to the database.
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+    async createRequestForAddingSavedDataset(request: Request, response: Response) {
+        let userEmail = request.params.userEmail;
+        let requestParam = request.params.datasetId;
+        let datasetId: number = +requestParam;
+        if (isNaN(datasetId)) {
+            response.status(400).json("Invalid data set ID entered");
+        }
+        else {
+            try {
+                let executionStatus = await this.dataSetService.addSavedDatasetService(userEmail, datasetId)
+                if (executionStatus[0]) { return response.status(200).json(executionStatus[1]); }
+                else { return response.status(400).json(executionStatus[1]); }
+            } catch (err) {
+                response.status(500).json(err);
+            }
+        }
+    }
+     */
+
+    /**
+     * This controller will take a request, send it to the getDataService to acquire an array 
+     * containing the unapproved data sets in the database
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUnapprovedDatsets(request: Request, response: Response) {
+        try {
+            this.dataSetService = new DataSetService();
+            let requestResponse = await this.dataSetService.getUnapprovedAllDatasets()
+            return response.status(requestResponse.statusCode).json(requestResponse.message);
+        } catch (error) {
+            response.status(error.status).json(error.message);
+        }
+    }
+
+    /**
+     * This method verifies that a request has at least one valid search parameter and returns true if so
+     * and false if not.
+     * 
+     * @param request 
+     * An object containing the request information and parameters: Request
+     */
+    private validateInputData(request: Request) {
+        if (request.query.hasOwnProperty('datasetId') || request.query.hasOwnProperty('material')

```

Comment:
i think in the future we could replace these checks with Yup

Reply:
Already planned